src/Symfony/Component/Serializer/Exception/AggregatedException.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Exception;
+
+class AggregatedException extends RuntimeException implements AggregableExceptionInterface
+{
+    /**
+     * @var array
+     */
+    private $collection;
+
+    public function __construct(\Throwable $previous = null)
+    {
+        parent::__construct('Errors occurred during the denormalization process', 0, $previous);
+    }
+
+    public function has(): bool
+    {
+        return !empty($this->collection);
+    }
+
+    public function add(string $param, AggregableExceptionInterface $exception): self
+    {
+        $this->collection[] = [$param, $exception];
+
+        return $this;
+    }
+
+    /**
+     * @return \Generator<string, AggregableExceptionInterface>
+     */
+    public function getExceptions(): \Generator
+    {
+        foreach ($this->collection as [$param, $exception]) {
+            if ($exception instanceof self) {
+                foreach ($exception->getExceptions() as $nestedParams => $nestedException) {
+                    yield "$param.$nestedParams" => $nestedException;
+                }
+            } else {
```

Comment:
```suggestion
                return;
            }
```

(to reduce the cyclomatic complexity)

Reply:
That doesn't work.  There is a composite pattern, thus it has to have two branch - for a leaf and for a node.

Reply:
Can we use `continue` then?

Reply:
Yes, we can. Updated